src/Modularity.Core/Loaders/FileModuleLoader.cs

```diff
@@ -51,43 +53,57 @@ private IEnumerable<IModule> EnumerateAssembliesByConfig(string directory, strin
 
             foreach (var config in configs.Where(x => x.IsActive))
             {
-                Module module;
-                try
+                Module module = new Module
                 {
-                    module = new Module
-                    {
-                        Assemblies = new List<Assembly>(),
-                        Name = config.Name,
-                        Exception = null,
-                        EntryObjects = new List<IEntryObject>()
-                    };
-
-                    var files = config.Files;
-                    if (files == null || files.Length == 0)
-                        files = Directory.GetFiles(Path.Combine(directory, config.Name), "*.dll");
+                    Assemblies = new List<Assembly>(),
+                    Name = config.Name,
+                    Exception = null,
+                    EntryObjects = new List<IEntryObject>()
+                };
 
-                    foreach (var file in files)
+                try
+                {
+                    if (config.Files != null && config.Files.Length > 0)
+                        foreach (var file in config.Files)
+                            try
+                            {
+                                using (var ms = new MemoryStream(File.ReadAllBytes(file)))
+                                    AssemblyLoadContext.Default.LoadFromStream(ms);
+                            }
+                            catch (Exception ex)
+                            {
+                                Debug.WriteLine(ex.ToString());
+                                module.Exceptions?.Add(ex);
+                            }
+
+                    var mainAssemblyName = Path.Combine(directory, config.Name, config.Name + ".dll");
+
+                    foreach (var file in Directory.GetFiles(Path.Combine(directory, config.Name), "*.dll"))
                     {
+                        var isMainAssembly = file == mainAssemblyName;
+                        if (!config.LoadAllDependencies && !isMainAssembly)
```

Comment:
If 	`LoadAllDependencies` is **false** in colfig, do not load anything else but main assembly.